src/PostgREST/Auth/JwtCache.hs

```diff
@@ -58,8 +59,13 @@ lookupJwtCache JwtCacheState{jwtCache} token maxLifetime parseJwt utc = do
 
       let timeSpec = getTimeSpec res maxLifetime utc
 
-      -- purge expired cache entries
-      C.purgeExpired jwtCache
+      -- trigger asynchronous purging of expired cache entries
+      -- but only if not already running anotherpurging
+      tryPutMVar purgeLock () >>= \case
+        -- start purge make sure lock is released
+        True -> void $ forkFinally (C.purgeExpired jwtCache) (const $ takeMVar purgeLock)
+        -- purge already running - do nothing
+        _ -> pure ()
```

Comment:
Finally, call the funtion here:

```suggestion
      purgeWithDebouce jwtCacheState
```

Reply:
Sorry, I don't understand why you think we are "creating" the debouncer lazily? `IO` actions are evaluated strictly. Also, we are initializing the it with `newEmptyMVar` in `init`.